### PR TITLE
Fix compile failed without gcc

### LINKED ISSUE #3129: Compiling without gcc
- Gitea version (or commit ref): b8e45b1e19895966617fe6db049de1a1af499b03
- Git version: 2.7.4
- Operating system: Ubuntu 16.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist: N/A

## Description
It should be possible to compile without gcc when building without sqlite support. Currently

```
jonas@be40f2203672:~/go/src/code.gitea.io/gitea# go build main.go
# code.gitea.io/gitea/vendor/github.com/mattn/go-sqlite3
exec: "gcc": executable file not found in $PATH
```

Related to: Issue https://github.com/go-gitea/gitea/issues/2172 and PR https://github.com/go-gitea/gitea/pull/2177

### PR DESCRIPTION
**Pull request by lunny**
will fix #3129

### CODE REVIEW

**Comment by ethantkoenig** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -131,7 +131,8 @@ fmt-check:
 
 .PHONY: test
 test:
-	$(GO) test $(PACKAGES)
+	$(GO) build # test if go build succeed without sqlite support
```
Comment (ethantkoenig) [inline Makefile:134]: Why is this under `test`? IMO, the `make test` target should run unit tests, nothing more.

~Also, building without the sqlite tag already runs in CI (`make build`), so why is doing it again necessary?~

**Review by ethantkoenig — CHANGES_REQUESTED** [review]
Sorry, I should have looked more closely before giving a LGTM

File: Makefile (hunk 1 above)
Comment (lunny) [reply Makefile:134]: `make build` will always add `-tags=sqlite`. This is a test for building without sqlite. Since all test now always use `sqlite`.

Comment (ethantkoenig) [reply Makefile:134]: Okay, I see why building again, without the sqlite tag, is necessary. However, could we please not put this in the `test` make target, for the sake of keeping our make targets straightforward? One option would be to just run `make build` an additional time in CI (without the sqlite tag).

Comment (lunny) [reply Makefile:134]: Or Maybe a new section named `comiple-test` or `build-test`?

**Comment by lafriks** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by lafriks** [comment]
Sorry @ethantkoenig  did not see that you have not approved

**Pull request merged**